src/klein/_app.py

```diff
@@ -223,19 +274,25 @@ def index(request):
 
         @named("router for '" + url + "'")
         def deco(f):
+            # type: (KleinRoute) -> KleinRoute
             kwargs.setdefault('endpoint', f.__name__)
             if kwargs.pop('branch', False):
                 branchKwargs = kwargs.copy()
                 branchKwargs['endpoint'] = branchKwargs['endpoint'] + '_branch'
 
                 @modified("branch route '{url}' executor".format(url=url), f)
                 def branch_f(instance, request, *a, **kw):
+                    # type: (Any, IRequest, Any, Any) -> KleinRenderable
                     IKleinRequest(request).branch_segments = (
                         kw.pop('__rest__', '').split('/')
                     )
                     return _call(instance, f, request, *a, **kw)
 
-                branch_f.segment_count = segment_count
+                branch_f = cast(KleinRoute, branch_f)
```

Comment:
`mypy` will flag this an unnecessary when we add type hints to the `@modified` decorator, which I had some trouble with.